### PR TITLE
feat(auth): migrate Bitbucket auth provider to module package

### PR DESCRIPTION
**Pull request by pjungermann**
Migrate the Bitbucket auth provider to the new `@backstage/plugin-auth-backend-module-bitbucket-provider` module package.

Relates-to: #19476

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/modern-hounds-give.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-auth-backend-module-bitbucket-provider': minor
+'@backstage/plugin-auth-backend': minor
```
Comment (Rugvip) [inline .changeset/modern-hounds-give.md:3]:
```suggestion
'@backstage/plugin-auth-backend': patch
```

No breaking change there, at least there shouldn't be 😁

File: plugins/auth-backend/api-report.md [hunk 2]
```diff
@@ -78,33 +78,6 @@ export type AuthResponse<TProviderInfo> = ClientAuthResponse<TProviderInfo>;
...
-  avatarUrl?: string;
-  _json?: {
-    links?: {
-      avatar?: {
-        href?: string;
-      };
-    };
-  };
-};
-
```
Comment (Rugvip) [inline plugins/auth-backend/api-report.md:107]: Let's just deprecate these for now

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice, thank you! 👍 

Just some stability bits and needs a rebase, but otherwise good to go

File: .changeset/modern-hounds-give.md (hunk 1 above)
Comment (pjungermann) [reply .changeset/modern-hounds-give.md:3]: alright, I will take care of that, and then it will be just a patch change, yes.

Comment (pjungermann) [reply .changeset/modern-hounds-give.md]: I have resolved both

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 

Got a conflict to resolve before we're able to merge

**Pull request merged**